### PR TITLE
Expose version as prometheus metric

### PR DESCRIPTION
**Pull request by jimmidyson**

### CODE REVIEW

**Comment by jimmidyson** [comment]
An unrelated e2e flake from the look of the logs...

**Comment by jimmidyson** [comment]
So 3 flakes... I can't see that they're related to the changes I've made.

**Comment by erictune** [comment]
@kubernetes/goog-gke @kubernetes/goog-cluster

**Comment by alex-mohr** [comment]
@fabioy want to take a quick look at this?

**Comment by fabioy** [comment]
Pretty straightforward. LGTM.

File: pkg/version/version.go [hunk 1]
```diff
@@ -45,3 +47,17 @@ func Get() Info {
...
+		prometheus.GaugeOpts{
+			Name: "kubernetes_build_info",
+			Help: "A metric with a constant '1' value labeled by major, minor, git version, git commit and git tree state from which Kubernetes was built.",
+		},
+		[]string{"major", "minor", "gitVersion", "gitCommit", "gitTreeState"},
+	)
+	info := Get()
+	buildInfo.WithLabelValues(info.Major, info.Minor, info.GitVersion, info.GitCommit, info.GitTreeState).Set(1)
+
+	prometheus.MustRegister(buildInfo)
```
Comment (fabioy) [inline pkg/version/version.go]: One small nit -- shouldn't you register the metric first before setting a value?

Comment (jimmidyson) [inline pkg/version/version.go]: Doesn't actually matter as this is a static metric. Normally you would though yes.

Comment (jimmidyson) [inline pkg/version/version.go]: This is exactly how Prometheus itself does it - see https://github.com/prometheus/prometheus/blob/master/version/info.go#L39

**Comment by alex-mohr** [comment]
@k8s-bot test this again please

File: pkg/version/version.go (hunk 1 above)
Comment (fabioy) [inline pkg/version/version.go]: Ha ha...well, there you go. Still feels a bit odd to me, but if it works...

**Comment by alex-mohr** [comment]
@k8s-bot test this please
May be a flake, but I don't see other similar failures. 

Jenkins:
Failed
integration.TestUnschedulableNodes (from k8s.io_kubernetes_test_integration)\

Error Message
Failed

Stacktrace
scheduler_test.go:264: Test 0: Pod did not get scheduled on an unschedulable node
scheduler_test.go:277: Test 0: failed to schedule a pod: timed out waiting for the condition
scheduler_test.go:264: Test 1: Pod did not get scheduled on an unschedulable node
scheduler_test.go:279: Test 1: Pod got scheduled on a schedulable node

**Comment by jimmidyson** [comment]
Another different failure... this cannot have any effect on these test failures. Well not that I can think of anyway - it would be extremely worrying if a minor change like this could stop pods being scheduled.

**Comment by wojtek-t** [comment]
@k8s-bot test this please

**Comment by alex-mohr** [comment]
@jimmidyson Apologies for the current spate of test infrastructure issues -- I agree those various different failures aren't due to your PR.  We're working to get those fixed and appreciate your patience.

@k8s-bot test this one more time

**Comment by jimmidyson** [comment]
@alex-mohr No worries! Hope you get it sorted soon.

**Comment by jimmidyson** [comment]
Phew!

**Pull request merged**